FSI/Displacement.C

```diff
@@ -41,23 +42,29 @@ void preciceAdapter::FSI::Displacement::read(double * buffer, const unsigned int
     {
         int patchID = patchIDs_.at(j);
 
+        const volVectorField &tmp_field = mesh_.lookupObject<volVectorField>("cellDisplacement");
+        vectorField           cellDisplacement(tmp_field.boundaryField()[patchID]);
+        forAll(cellDisplacement, i)
+        {
+            // Set the displacement to the received one
+            cellDisplacement[i][0] = buffer[bufferIndex++];
+            cellDisplacement[i][1] = buffer[bufferIndex++];
+            if(dim ==3)
+                cellDisplacement[i][2] = buffer[bufferIndex++];
+        }
+
         // Get the displacement on the patch
-        fixedValuePointPatchVectorField& pointDisplacementFluidPatch
+        vectorField& pointDisplacementFluidPatch
         (
-            refCast<fixedValuePointPatchVectorField>
+            refCast<vectorField>
             (
                 pointDisplacement_->boundaryFieldRef()[patchID]
             )
         );
 
-        // For every cell of the patch
-        forAll(pointDisplacement_->boundaryFieldRef()[patchID], i)
-        {
-            // Set the displacement to the received one
-            pointDisplacementFluidPatch[i][0] = buffer[bufferIndex++];
-            pointDisplacementFluidPatch[i][1] = buffer[bufferIndex++];
-            if(dim ==3)
-                pointDisplacementFluidPatch[i][2] = buffer[bufferIndex++];
-        }
+        //Interpolate from centers to nodes
+        primitivePatchInterpolation patchInterpolator(mesh_.boundaryMesh()[patchID]);
+        pointDisplacementFluidPatch =
+             patchInterpolator.faceToPointInterpolate(cellDisplacement);
```

Comment:
> Do I understand this change correctly?
> 
> To read Displacement:
> 
>     * Before: get the `boundaryFieldRef` of `pointDisplacement` and assign it buffer values.

Correct. 
> 
>     * Now: get the `boundaryField` (why?) of  `cellDisplacement` and assign it buffer values. Then, interpolate the `cellDisplacement` from the face centers to `pointDisplacement` on the face nodes.
> 
It's probably a confusing choice. I just want to use an appropriate constructor for the temporary `volVectorField` in order to allocate the correct size (it's not a reference but a copy). Every other `volVectorField` would be good as well. The `cellDisplacement` was just an obvious choice. There is probably a better way of doing this.

>     * Additionally: replace the more specific type `fixedValuePointPatchVectorField` with the more general `vectorField` (why?)

Compatibility with the `primitivePatchInterpolation` object. Otherwise I need another temporary field in order to assign values from the interpolated temporary `volVectorField` to the `fixedValuePointPatchVectorField`.